lib/Parse/Lexer.cpp

```diff
@@ -422,26 +425,29 @@ void Lexer::skipSlashStarComment() {
 
       break;   // Otherwise, eat other characters.
     case 0:
-      // If this is a random nul character in the middle of a buffer, skip it as
-      // whitespace.
-      if (CurPtr-1 != BufferEnd) {
-        diagnoseEmbeddedNul(Diags, CurPtr-1);
-        break;
-      }
-      
-      // Otherwise, we have an unterminated /* comment.
-      --CurPtr;
+      switch (getNulCharacterKind(CurPtr - 1)) {
+      case NulCharacterKind::Embedded:
+        // If this is a random nul character in the middle of a buffer, skip it
+        // as whitespace.
+        diagnoseEmbeddedNul(Diags, CurPtr - 1);
+        LLVM_FALLTHROUGH;
+      case NulCharacterKind::CodeCompletion:
+        continue;
+      case NulCharacterKind::BufferEnd:
```

Comment:
The last request from me.
Please enclose this branch with brace `case NulCharacterKind::BufferEnd: { ... }`. Lacking braces make us difficult to add another `case` (or re-order `case`s).